pkgs/misc/uboot/default.nix

```diff
@@ -366,6 +391,103 @@ in {
+      })
+      (fetchpatch {
+        url = "https://raw.githubusercontent.com/armbian/build/420f16c7ead971f1c443314ef41d9a02473cf260/patch/u-boot/u-boot-rockchip64/general-set-eth1addr.patch";
+        sha256 = "sha256-u/kiWf0vz3Yw+rEgAO+BPw/PkKom3SM2F9jOS/6/Nu0=";
+      })
+        (fetchpatch {
+        url = "https://raw.githubusercontent.com/armbian/build/420f16c7ead971f1c443314ef41d9a02473cf260/patch/u-boot/u-boot-rockchip64/general-support-recovery-button.patch";
+        sha256 = "sha256-uP1eLOALtxS7Ue9heVAkYFDwvW0lTZdQfhVioh5oBNc=";
+      })
+      (fetchpatch {
+        url = "https://raw.githubusercontent.com/armbian/build/420f16c7ead971f1c443314ef41d9a02473cf260/patch/u-boot/u-boot-rockchip64/general-support-rmii-integrated-phy.patch";
+        sha256 = "sha256-y4FaQ1OQGVvkD8Sot0jNJMuGO6p9o2u03JiL5MKihm4=";
+      })
+      (fetchpatch {
+        url = "https://raw.githubusercontent.com/armbian/build/420f16c7ead971f1c443314ef41d9a02473cf260/patch/u-boot/u-boot-rockchip64/rk3328-efuse-driver.patch";
+        sha256 = "sha256-ly2dJb1BRXuITfNYCiqqIxqErWQnrdonhpaLwmeKAm4=";
+      })
+      (fetchpatch {
+        url = "https://raw.githubusercontent.com/armbian/build/420f16c7ead971f1c443314ef41d9a02473cf260/patch/u-boot/u-boot-rockchip64/rk3328-resets-for-mmc-controllers.patch";
+        sha256 = "sha256-abd2V0D2Cm5hw+JiWa80c7kmhcV8G6wbyyxw/ZAWu+0=";
+      })
+      (fetchpatch {
+        url = "https://raw.githubusercontent.com/armbian/build/420f16c7ead971f1c443314ef41d9a02473cf260/patch/u-boot/u-boot-rockchip64/rk3328-sdmmc_ext-node.patch";
+        sha256 = "sha256-EHS/aBik4V4pbEU9GHhCXCBgdoExbVvUzB01KDx4zM8=";
+      })
+      (fetchpatch {
+        url = "https://raw.githubusercontent.com/armbian/build/420f16c7ead971f1c443314ef41d9a02473cf260/patch/u-boot/u-boot-rockchip64/sdmmc-force-fifo-mode-in-spl.patch";
+        sha256 = "sha256-+oy6kG/wAvd89M+ZS6a2xlKWzzdHV6UCtd1SaTmLJaI=";
+      })
+    ];
+    extraMeta = {
+      platforms = [ "aarch64-linux" ];
+      license = lib.licenses.unfreeRedistributableFirmware;
+    };
+    BL31 = "${armTrustedFirmwareRK3328}/bl31.elf";
+    filesToInstall = [ "u-boot.img" "idbloader.img" "trust.bin" ];
+    postBuild = ''
+      # deduced from https://github.com/armbian/build/blob/420f16c7ead971f1c443314ef41d9a02473cf260/config/boards/orangepi-r1plus-lts.conf#L4
+      # and https://github.com/armbian/build/blob/420f16c7ead971f1c443314ef41d9a02473cf260/config/sources/families/include/rockchip64_common.inc#L42
+      BOOT_SOC=rk3328
+
+      # see https://github.com/orangepi-xunlong/orangepi-build/blob/f00cd197b4a9873f36093d4f4748b733642059a7/external/config/sources/families/include/rockchip64_common.inc#L29-L35
+      # or rather https://github.com/armbian/build/blob/420f16c7ead971f1c443314ef41d9a02473cf260/config/sources/families/include/rockchip64_common.inc#L70-L75
+      DDR_BLOB=${rkbin}/rk33/rk3328_ddr_333MHz_v1.16.bin
```

Comment:
Ram training / TPL may or may not be available from mainline U-Boot.

In any case, it would have to be tested first to be sure it works compared to the blob.

So even in a follow-up relying solely on U-Boot, it would be fine to have this blob.